renderdoc/driver/vulkan/vk_info.cpp

```diff
@@ -446,6 +446,19 @@ void DescSetLayout::CreateBindingsArray(BindingStorage &bindingStorage, uint32_t
 
       bindingStorage.inlineBytes.resize(inlineByteSize);
     }
+
+    for(size_t i = 0; i < bindings.size(); i++)
+    {
+      if(!bindings[i].immutableSampler)
+        continue;
+
+      for(uint32_t j = 0; j < bindings[i].descriptorCount; j++)
+      {
+        DescriptorSetSlot *slot = bindingStorage.binds[i] + j;
+        slot->type = convert(bindings[i].layoutDescType);
+        slot->sampler = bindings[i].immutableSampler[j];
+      }
+    }
```

Comment:
This affects much more than shader debugging. I see an issue with the shader debug code but it should be fixed there, rather than here which has knock-on effects.